hazelcast/src/main/java/com/hazelcast/query/QueryableEntry.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.query;
+
+import java.util.Map;
+
+/**
+ * Represents a Queryable Entry, which means searched an indexed by SQL query or predicate.
+ * @param <K></K> the key type
+ * @param <V></V> the value type
+ */
+public interface QueryableEntry<K, V> extends Map.Entry<K, V> {
```

Comment:
Can we come up with a better name for this?

Reply:
We can probably get rid of this interface altogether and stick with just `Map.Entry` on the public API level. The reason why I introduced this interface was to keep our hands free if we need to add additional methods, extend another interface etc.